runtime/vm/classsupport.c

```diff
@@ -108,7 +108,7 @@ internalFindArrayClass(J9VMThread* vmThread, J9Module *j9module, UDATA arity, U_
 		/* the first level of arity is already present in the array class */
 		arity -= 1;
 
-	} else if (firstChar == 'L' && lastChar == ';') {
+	} else if ((firstChar == 'L' || firstChar == 'Q') && lastChar == ';') {
```

Comment:
one more thing, we need a #ifdef VT here. Define a macro like this in j9.h (youll see the other VT macros)

```
#ifdef VT
#define IS_REFERENCE_SIGNATURE(firstChar) ('L' == (firstChar))
#else
#define IS_REFERENCE_SIGNATURE(firstChar) ('L' == (firstChar) || 'Q' == (firstChar))
#endif
```

Reply:
perhaps a better name is `IS_REF_OR_VAL_SIGNATURE`